### ISSUE TITLE
BUG: parsing boolean raises TypeError without format, ValueError with format

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#50584]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [1]: to_datetime([False], format='%Y-%m-%d')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 1
----> 1 to_datetime([False], format='%Y-%m-%d')

File ~/pandas-dev/pandas/core/tools/datetimes.py:1063, in to_datetime(arg, errors, dayfirst, yearfirst, utc, format, exact, unit, infer_datetime_format, origin, cache)
   1061         result = _convert_and_box_cache(argc, cache_array)
   1062     else:
-> 1063         result = convert_listlike(argc, format)
   1064 else:
   1065     result = convert_listlike(np.array([arg]), format)[0]

File ~/pandas-dev/pandas/core/tools/datetimes.py:435, in _convert_listlike_datetimes(arg, format, name, utc, unit, errors, dayfirst, yearfirst, exact)
    432     format = _guess_datetime_format_for_array(arg, dayfirst=dayfirst)
    434 if format is not None:
--> 435     return _array_strptime_with_fallback(arg, name, utc, format, exact, errors)
    437 result, tz_parsed = objects_to_datetime64ns(
    438     arg,
    439     dayfirst=dayfirst,
   (...)
    443     allow_object=True,
    444 )
    446 if tz_parsed is not None:
    447     # We can take a shortcut since the datetime64 numpy array
    448     # is in UTC

File ~/pandas-dev/pandas/core/tools/datetimes.py:466, in _array_strptime_with_fallback(arg, name, utc, fmt, exact, errors)
    455 def _array_strptime_with_fallback(
    456     arg,
    457     name,
   (...)
    461     errors: str,
    462 ) -> Index:
    463     """
    464     Call array_strptime, with fallback behavior depending on 'errors'.
    465     """
--> 466     result, timezones = array_strptime(arg, fmt, exact=exact, errors=errors, utc=utc)
    467     if any(tz is not None for tz in timezones):
    468         return _return_parsed_timezone_results(result, timezones, utc, name)

File ~/pandas-dev/pandas/_libs/tslibs/strptime.pyx:494, in pandas._libs.tslibs.strptime.array_strptime()
    492     continue
    493 elif is_raise:
--> 494     raise
    495 return values, []
    496 

File ~/pandas-dev/pandas/_libs/tslibs/strptime.pyx:323, in pandas._libs.tslibs.strptime.array_strptime()
    321 found = format_regex.match(val)
    322 if not found:
--> 323     raise ValueError(f"time data \"{val}\" doesn't "
    324                      f"match format \"{fmt}\"")
    325 if len(val) != found.end():

ValueError: time data "False" doesn't match format "%Y-%m-%d", at position 0

In [2]: to_datetime([False])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 to_datetime([False])

File ~/pandas-dev/pandas/core/tools/datetimes.py:1063, in to_datetime(arg, errors, dayfirst, yearfirst, utc, format, exact, unit, infer_datetime_format, origin, cache)
   1061         result = _convert_and_box_cache(argc, cache_array)
   1062     else:
-> 1063         result = convert_listlike(argc, format)
   1064 else:
   1065     result = convert_listlike(np.array([arg]), format)[0]

File ~/pandas-dev/pandas/core/tools/datetimes.py:437, in _convert_listlike_datetimes(arg, format, name, utc, unit, errors, dayfirst, yearfirst, exact)
    434 if format is not None:
    435     return _array_strptime_with_fallback(arg, name, utc, format, exact, errors)
--> 437 result, tz_parsed = objects_to_datetime64ns(
    438     arg,
    439     dayfirst=dayfirst,
    440     yearfirst=yearfirst,
    441     utc=utc,
    442     errors=errors,
    443     allow_object=True,
    444 )
    446 if tz_parsed is not None:
    447     # We can take a shortcut since the datetime64 numpy array
    448     # is in UTC
    449     dta = DatetimeArray(result, dtype=tz_to_dtype(tz_parsed))

File ~/pandas-dev/pandas/core/arrays/datetimes.py:2158, in objects_to_datetime64ns(data, dayfirst, yearfirst, utc, errors, allow_object)
   2156 order: Literal["F", "C"] = "F" if flags.f_contiguous else "C"
   2157 try:
-> 2158     result, tz_parsed = tslib.array_to_datetime(
   2159         data.ravel("K"),
   2160         errors=errors,
   2161         utc=utc,
   2162         dayfirst=dayfirst,
   2163         yearfirst=yearfirst,
   2164     )
   2165     result = result.reshape(data.shape, order=order)
   2166 except OverflowError as err:
   2167     # Exception is raised when a part of date is greater than 32 bit signed int

File ~/pandas-dev/pandas/_libs/tslib.pyx:441, in pandas._libs.tslib.array_to_datetime()
    439 @cython.wraparound(False)
    440 @cython.boundscheck(False)
--> 441 cpdef array_to_datetime(
    442     ndarray[object] values,
    443     str errors="raise",

File ~/pandas-dev/pandas/_libs/tslib.pyx:645, in pandas._libs.tslib.array_to_datetime()
    643 
    644     except TypeError:
--> 645         return _array_to_datetime_object(values, errors, dayfirst, yearfirst)
    646 
    647     if seen_datetime_offset and not utc_convert:

File ~/pandas-dev/pandas/_libs/tslib.pyx:778, in pandas._libs.tslib._array_to_datetime_object()
    776     else:
    777         if is_raise:
--> 778             raise
    779         return values, None
    780 return oresult, None

File ~/pandas-dev/pandas/_libs/tslib.pyx:629, in pandas._libs.tslib.array_to_datetime()
    627             iresult[i] = NPY_NAT
    628         else:
--> 629             raise TypeError(f"{type(val)} is not convertible to datetime")
    630 
    631 except OutOfBoundsDatetime as ex:

TypeError: <class 'bool'> is not convertible to datetime
```

### Issue Description

The error raised is inconsistent

### Expected Behavior

Just raise `ValueError` in both cases?

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8acd31495ab2c0baaaac466089a109d8a637a061
python           : 3.8.16.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.10.102.1-microsoft-standard-WSL2
Version          : #1 SMP Wed Mar 2 00:30:59 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 2.0.0.dev0+1049.g8acd31495a
numpy            : 1.23.5
pytz             : 2022.7
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 22.3.1
Cython           : 0.29.32
pytest           : 7.2.0
hypothesis       : 6.61.0
sphinx           : 5.3.0
blosc            : 1.11.1
feather          : None
xlsxwriter       : 3.0.6
lxml.etree       : 4.9.2
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : 2.9.5
jinja2           : 3.1.2
IPython          : 8.8.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : 1.3.5
brotli           : 
fastparquet      : 2022.12.0
fsspec           : 2022.11.0
gcsfs            : 2022.11.0
matplotlib       : 3.6.2
numba            : 0.56.4
numexpr          : 2.8.4
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : 1.2.0
pyxlsb           : 1.0.10
s3fs             : 2022.11.0
scipy            : 1.10.0
snappy           : 
sqlalchemy       : 1.4.46
tables           : 3.8.0
tabulate         : 0.9.0
xarray           : 2022.12.0
xlrd             : 2.0.1
zstandard        : 0.19.0
tzdata           : 2022.7
qtpy             : None
pyqt5            : None
None

</details>

**Comment by Mateko** [comment]
take

**Comment by MarcoGorelli** [comment]
ah this was fixed (and tested for) in #50586, forgot to close, sorry
